src/core/exporter.js

```diff
@@ -0,0 +1,104 @@
+// module: exporter
+// used in ui/save-html
+
+import { removeReSpec } from "core/utils";
+import { pub } from "core/pubsubhub";
+
+export function serialize(format = "html", doc = document) {
+  const cloneDoc = doc.cloneNode(true);
+  cleanup(cloneDoc);
+  let result = "";
+  switch (format) {
+    case "xml":
+      result = new XMLSerializer().serializeToString(cloneDoc);
+      break;
+    default: {
+      if (cloneDoc.doctype) {
+        result += new XMLSerializer().serializeToString(cloneDoc.doctype);
+      }
+      result += cloneDoc.documentElement.outerHTML;
+    }
+  }
+  return result;
+}
+
+export async function exportDocument(format, mimeType, doc = document) {
```

Comment:
We should take this opportunity to document this function.

Reply:
The more I think about it, `exportDocument()` should be the only thing we export from this module.

Reply:
yes. we'll make `exportDocument` smarter